lib/oauth2/access_token.rb

```diff
@@ -11,7 +11,9 @@ class << self
       # @return [AccessToken] the initalized AccessToken
       def from_hash(client, hash)
         hash = hash.dup
-        new(client, hash.delete('access_token') || hash.delete(:access_token), hash)
+        token =  hash.delete('access_token') || hash.delete(:access_token)
```

Comment:
I'm thinking at this point, maybe we use an option from client like https://github.com/oauth-xx/oauth2/pull/511/files#diff-bc9331245a41740ee0145e2d5640ed07R29 but not limit to 1 but multiple mapping. i.e. 

```ruby
{
  access_token: ['access_token', 'id_token']
}
```

Then try each mapping. 
For Pinterest
```ruby
{
  access_token: 'data,access_token' # OR access_token: ['data,access_token']
}
```

Then 
```ruby
token = nil
client.response_mapping[:access_token].detect do |mapping|
  token = hash.dig(mapping.split(','))
end
```

What do you think? @jonspalmer

Reply:
I don't think a single option for the "path" to dig is enough. Not only do you want to get the access token but you also need to remove it from the hash before actually constructing the `AccessToken` etc. Various customized clients might have reason to otherwise manipulate the hash etc.

Reply:
So, how about support both? :)